grpc-api/src/main/java/io/deephaven/grpc_api/table/ops/AsOfJoinTablesGrpcImpl.java

```diff
@@ -0,0 +1,88 @@
+package io.deephaven.grpc_api.table.ops;
+
+import com.google.common.collect.Lists;
+import com.google.rpc.Code;
+import io.deephaven.base.verify.Assert;
+import io.deephaven.db.exceptions.ExpressionException;
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.tables.live.LiveTableMonitor;
+import io.deephaven.db.tables.select.MatchPair;
+import io.deephaven.db.tables.select.MatchPairFactory;
+import io.deephaven.grpc_api.session.SessionState;
+import io.deephaven.grpc_api.util.GrpcUtil;
+import io.deephaven.proto.backplane.grpc.AsOfJoinTablesRequest;
+import io.deephaven.proto.backplane.grpc.BatchTableRequest;
+import io.deephaven.proto.backplane.grpc.JoinTablesRequest;
+import io.deephaven.util.FunctionalInterfaces;
+import io.grpc.StatusRuntimeException;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class AsOfJoinTablesGrpcImpl extends GrpcTableOperation<AsOfJoinTablesRequest> {
```

Comment:
should be singleton

Reply:
I haven't reviewed the grpc layer extensively, but looking at the code, it doesn't *need* to be a singleton, does it? (I'm looking at GrpcTableOperations, and it doesn't look like it keeps any state.)